### PR TITLE
Fix how apm is started in cloud legacy mode

### PR DESCRIPTION
**Pull request by michalpristas**
Mainly changing input type and setting proper working directory

Do not merge, i'm going to merge after checking with apm team

### CODE REVIEW

File: internal/pkg/agent/cmd/container.go [hunk 1]
```diff
@@ -701,8 +701,8 @@ func truncateString(b []byte) string {
 
 // runLegacyAPMServer extracts the bundled apm-server from elastic-agent
 // to path and runs it with args.
-func runLegacyAPMServer(streams *cli.IOStreams, path string) (*process.Info, error) {
-	name := "apm-server"
+func runLegacyAPMServer(streams *cli.IOStreams) (*process.Info, error) {
+	name := "apm"
```
Comment (simitt) [inline internal/pkg/agent/cmd/container.go:705]: why did this change and does it have any consequences on the path, log path, etc?

**Review by AndersonQ — APPROVED** [review]
Is there any test that checks the agent can start the legacy apm server?

File: internal/pkg/agent/cmd/container.go (hunk 1 above)
Comment (michalpristas) [reply internal/pkg/agent/cmd/container.go:705]: no this is for our resolution. based on this name we find correct process. while before it was a binary name now we have a mapping for input type

**Comment by michalpristas** [comment]
i'm going to merge this. testing locally is resource heavy and i failed to verify this solves the issue, this getting merged will not break it even more than it is and gives me some more space for testing

**Pull request merged**